src/background/permissions-state-tracker.ts

```diff
@@ -0,0 +1,35 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+import { Interpreter } from 'background/interpreter';
+import { BrowserAdapter } from 'common/browser-adapters/browser-adapter';
+import { Message } from 'common/message';
+import { Messages } from 'common/messages';
+
+export class PermissionsStateTracker {
```

Comment:
Thinking about this name more, maybe `BrowserPermissionsTracker` would be a more descriptive name?